src/azure-cli-core/azure/cli/core/commands/client_factory.py

```diff
@@ -95,20 +95,26 @@ def configure_common_settings(cli_ctx, client):
     except KeyError:
         pass
 
+    # Prepare CommandName header
+    command_name_suffix = ';completer-request' if cli_ctx.data['completer_active'] else ''
+    cli_ctx.data['headers']['CommandName'] = "{}{}".format(cli_ctx.data['command'], command_name_suffix)
+
+    # Prepare ParameterSetName header
+    if cli_ctx.data.get('safe_params'):
+        cli_ctx.data['headers']['ParameterSetName'] = ' '.join(cli_ctx.data['safe_params'])
```

Comment:
Refactored the original code to save headers in `cli_ctx.data['headers']` and call `client._client.add_header` in one place.